### PR TITLE
GitHub actions: concurrent, independent steps

### PR DESCRIPTION
**Pull request by jgehrcke**
This addresses https://github.com/conbench/conbench/issues/554.

Before:
image

After:
image

### CODE REVIEW

File: conbench/tests/populate_local_conbench.py [hunk 1]
```diff
@@ -196,21 +219,39 @@ def register():
...
+            break
+        except requests.exceptions.RequestException as exc:
+            log.info(
+                "attempt %s failed with %s after %.5f",
+                attempt,
+                exc,
+                time.monotonic() - t0,
+            )
+            time.sleep(5 * attempt)
+
```
Comment (jgehrcke) [inline conbench/tests/populate_local_conbench.py:249]: This was to debug and address https://github.com/conbench/conbench/issues/555.

**Comment by jgehrcke** [comment]
`ci/testsuite` failed with known instability: https://github.com/conbench/conbench/issues/502
https://github.com/conbench/conbench/actions/runs/3749721967/jobs/6368519481#step:3:1725

I think this looks good, merging. To add screenshotty things.

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -45,6 +42,61 @@ jobs:
...
+  # This is to confirm that commonly used developer commands still work.
+  dev-cmds:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
```
Comment (austin3dickey) [inline .github/workflows/actions.yml:55]: The first time I realized that YML interprets `python-version: 3.10` (with no quotation marks) as the number equivalent to "3.1" I laughed out loud.

File: conbench/tests/populate_local_conbench.py [hunk 3]
```diff
@@ -196,21 +219,39 @@ def register():
...
+        log.info("POST to url: %s", url)
+        try:
+            # Often seeing RemoteDisconnected when the processing
+            # takes too long. Also see
+            # https://github.com/conbench/conbench/issues/555
+            res = session.post(url, json=data)
+            break
+        except requests.exceptions.RequestException as exc:
+            log.info(
+                "attempt %s failed with %s after %.5f",
```
Comment (jgehrcke) [inline conbench/tests/populate_local_conbench.py:243]: Oh, I forgot a ` s` in the string :), the unit.

File: .github/workflows/actions.yml (hunk 2 above)
Comment (jgehrcke) [reply .github/workflows/actions.yml:55]: Indeed rather surprising and ridiculous. This is the second or third time I run into this, too.

Thanks for looking!

**Pull request merged**